### PR TITLE
Only attempt to record an exception if there is an active span.

### PR DESCRIPTION
**Pull request by derekkraan**
There is an edge case, if you use `forward/4` and use Plug.ErrorHandler,
then when an exception reaches the outer router, then Plug.send_resp
will be called, triggering `[:phoenix, :endpoint, :stop]`, and the span
will be gone by the time the outer router gets the exception. This
causes this telemetry handler to crash and be detached.

Sequence of events:
- [:phoenix, :endpoint, :start]
- [:phoenix, :router_dispatch, :exception] (inner router)
- [:phoenix, :endpoint, :stop]
- [:phoenix, :router_dispatch, :exception] (outer router) ** here there is no span, crashes

### What does this PR do?

This PR fixes an edge case that results in the exception telemetry event handler being detached.

### CODE REVIEW

**Review by bryannaegele — DISMISSED** [review]

**Comment by bryannaegele** [comment]
Looks like you need to format @derekkraan

**Comment by derekkraan** [comment]
@bryannaegele formatted and now CI is happy.

**Comment by derekkraan** [comment]
@bryannaegele is it possible to get this merged?

File: instrumentation/opentelemetry_phoenix/lib/opentelemetry_phoenix.ex [hunk 1]
```diff
@@ -162,21 +162,23 @@ defmodule OpentelemetryPhoenix do
...
-    exception = Exception.normalize(kind, reason, stacktrace)
+      # try to normalize all errors to Elixir exceptions
+      exception = Exception.normalize(kind, reason, stacktrace)
 
-    # record exception and mark the span as errored
-    Span.record_exception(ctx, exception, stacktrace, attrs)
-    Span.set_status(ctx, OpenTelemetry.status(:error, ""))
+      # record exception and mark the span as errored
+      Span.record_exception(ctx, exception, stacktrace, attrs)
+      Span.set_status(ctx, OpenTelemetry.status(:error, ""))
```
Comment (bryannaegele) [inline instrumentation/opentelemetry_phoenix/lib/opentelemetry_phoenix.ex:177]: Looks like there's some kind of whitespace thing going on here?

Comment (derekkraan) [reply instrumentation/opentelemetry_phoenix/lib/opentelemetry_phoenix.ex:177]: There is a big `if` statement around the whole thing, that's where the extra indenting is coming from.

**Comment by bryannaegele** [comment]
@derekkraan can you confirm maintainers have edit access on your PR?

**Comment by derekkraan** [comment]
@bryannaegele I can't seem to find that option anywhere.

**Comment by derekkraan** [comment]
I'm looking here but I don't see the checkbox https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork

**Comment by bryannaegele** [comment]
I think it's only there when you open it. I'm trying to improve my workflow to be less onerous.

Would you mind closing and reopening with that checked just so I can try it again?

**Comment by derekkraan** [comment]
Yeah I'll give it a proper go tomorrow. I think I'll need a new branch to properly re-open it.

**Comment by bryannaegele** [comment]
You should be able to just close and then hit open a PR on your fork.

**Comment by derekkraan** [comment]
@bryannaegele I tried it but I honestly can not find this checkbox, even on the new PR form.
image

**Comment by derekkraan** [comment]
@bryannaegele we added you to our repo as a collaborator, apparently that's a poorly-documented necessity in order to get what you want.

**Comment by tsloughter** [comment]
Is this good to merge now?

**Review by tsloughter — APPROVED** [review]

**Review by bryannaegele — APPROVED** [review]

**Comment by bryannaegele** [comment]
@derekkraan seems that invite expired on me. Can you just rebase and I'll merge?

**Comment by derekkraan** [comment]
@bryannaegele rebased on `main`

**Pull request merged**